website/main/content/blog/2025-05-19-block-as-forest.md

```diff
@@ -0,0 +1,133 @@
+---
+title: The Unblocker - Blocks as a forest of trees
+date: 2025-05-19
+draft: false
+description: Unblocking progress through the design of blocks as a forest of trees.
+tags: [status-update, consensus]
+authors: [adlrocha]
+---
+
+I am pretty happy with the progress this week. Funnily, the main culprit for all of this progress
+has been the the work that Nazar has been doing on the definition of the block structure for our
+hierarchical consensus. I'll let Nazar dig deeper into what he's been doing here, but let me share
+in this post what this block structure entails, and how this has unblock several lines of work, and
+solved many issues for me.
+
+<!--more-->
+
+## Cracking the first piece of the puzzle: blocks as trees.
+
+If you recall from previous posts, we already had a pretty good sense of what is going to be the
+lifecycle of blocks in the system: from being created in a shard, to being submitted and committed
+to its parent, and how this information is implicitly propagated up and can be used to generate
+proofs about the history of a shard. Unfortunately, we were missing one piece to this puzzle that
+was dragging progress: making a decision of what would be the block structure, as this would
+influence the design of other mechanics in the system. After a few back-and-forths, it looks like
+Nazar has cracked it down on the PRs referenced by
+[this discussion](https://abundance.zulipchat.com/#narrow/channel/495788-research/topic/Mechanics.20of.20child.20block.20submission.20to.20parent.20chain/near/518994821).
```

Comment:
```suggestion
Nazar has cracked it down on the PRs referenced by [this discussion].
```

And then separately define a URL, I usually do that right after the paragraph where it is used:
```markdown
[this discussion]: Could be more convenient for formatting to separate URLs from their usage. For example here:
```

This results in a bit more readable source code and if URL is right next to it, doesn't cause difficulties when wanting to follow the link.